### ISSUE TITLE
About the last normalization layer 

### ISSUE THREAD

**Issue by bluenights** [facebookresearch/barlowtwins#40]
Thanks for the great work! Before computing the cross-correlation matrix, can we L2-normalize the representations along the batch dimension instead of using BatchNorm1d? Then the pseudocode becomes:

# normalize repr. along the batch dimension
z_a_norm = torch.nn.functional.normalize(z_a, dim=0) # NxD
z_b_norm = torch.nn.functional.normalize(z_b, dim=0) # NxD
# cross-correlation matrix
c = mm(z_a_norm.T, z_b_norm) # DxD

**Comment by zhangdan8962** [comment]
I believe L2 only ensures that the tensor has unit length, it has nothing to do with mean and std.
